### PR TITLE
display list item names

### LINKED ISSUE #1: 1. As a user, I want to read the names of all of the items in my shopping list
## Summary

The most important feature of our app is the ability to look at what's on the list, so the user knows what they need to buy. The `List` component is the best place in our code to do that. In fact, this feature is already partially implemented. Refer to the acceptance criteria in this issue, and the existing code, to understand the work that remains.

The `List` component is rendered on the `/list` page. Because the navigation links are currently broken, you'll have to type that path into your browser's address bar.

## Acceptance criteria
- [x] The `streamListItems` function is used in `App.jsx` to get the items in the current list from the Firestore database
- [x] The list is passed into the `List` component as a prop named `data`
- [ ] In `List.jsx`, the `ListItem` component is used to render the name of each item

## Notes
- ⚠️ Your **only goal** is to make sure that the names of the items render as expected. You might be tempted to modify `ListItem.jsx` to show more information – you will do that **later**. If you want a peek at the future, [this wireframe](https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png) is what your list could look like closer to the end of the project.
- The data you need is already being passed into the `List` component. What steps do you need to take in order to render it?

### PR DESCRIPTION
**Pull request by labrocadabro**
Co-authored-by: jtooba70@gmail.com

## Description

Adds the names of the list items to the /list page.

## Related Issue

Closes #1.

## Acceptance Criteria

 - The streamListItems function is used in App.jsx to get the items in the current list from the Firestore database
 - The list is passed into the List component as a prop named data
 - In List.jsx, the ListItem component is used to render the name of each item

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

### Before

List Page before change

### After

List Page after change

## Testing Steps / QA Criteria

View the /list page and see the list item names.

### CODE REVIEW

**Comment by labrocadabro** [comment]
We opted for name={item.name} to pass the name, as it's more aligned with the specific requirements of this PR, but I'm curious what mentors think of using {...item}? It doesn't change the visible outcome and still meets all the AC, but does prepare for known future changes that will need to be made. 

Interested in it from a best practice point of view - do you keep scope as limited as possible? I know you shouldn't write code to prepare for *unknown* future changes, but what about *known* ones? Do you commit code that you know will need to be replaced later, or let the scope of the solution expand slightly?

**Comment by shahx95** [comment]
Great job @labrocadabro @ToobaJamal ! The changes look good to me. Thanks! 🤗

**Comment by shahx95** [comment]
Since it wasn't mentioned in the acceptance criteria, I was wondering if it might be possible for the mentors to let us know if we should consider adding a unique "key" prop to each ListItem component in the future?

**Comment by ToobaJamal** [comment]
> 

Ah yes, good question. Might also be an obvious thing.
@paramsiddharth, @RajGM your thoughts?

**Comment by labrocadabro** [comment]
Oh right!  Personally I'd lean towards adding another commit to the PR that adds the key, since we're getting a console warning about the issue.

**Comment by paramsiddharth** [comment]
> We opted for name={item.name} to pass the name, as it's more aligned with the specific requirements of this PR, but I'm curious what mentors think of using {...item}? It doesn't change the visible outcome and still meets all the AC, but does prepare for known future changes that will need to be made.

I am all in for it — The only point to note here is that the `item's properties shouldn't collide with other HTML or component props!

> Interested in it from a best practice point of view - do you keep scope as limited as possible? I know you shouldn't write code to prepare for _unknown_ future changes, but what about _known_ ones? Do you commit code that you know will need to be replaced later, or let the scope of the solution expand slightly?

It's more than Schrödinger's cat when we speak of good practices, let me explain:
1. In most situations, keeping the scope as limited as possible is recommended. Why? Because most importantly, we want all props to be evidently visible so that we _know_ what we're sending in if it ever breaks in future.
2. Keeping the scope limited, both in Python's named arguments and React's props, comes at a cost of repetition. Nobody wants to write `prop1={prop1} prop2={prop2} prop3={prop3} prop4={prop4} prop5={prop5}...`. There's a limit to how much we focus on the scope being evident. Until that limit is crossed, we keep it explicit.
3. When that limit is crossed, however, we weigh towards modifying the data model to align with the needs of the component, and not the other way round. In that situation, the priority changes to keep the data model (here, `item`) tightly in line with the component's needs, and there we can simply go with `{...item}`.
4. In extremely simple and non-changing situations, we can omit overthinking and simply follow whatever suits us. :)

**Comment by paramsiddharth** [comment]
> Since it wasn't mentioned in the acceptance criteria, I was wondering if it might be possible for the mentors to let us know if we should consider adding a unique "key" prop to each ListItem component in the future?

> Ah yes, good question. Might also be an obvious thing. @paramsiddharth, @RajGM your thoughts?

> Oh right! Personally I'd lean towards adding another commit to the PR that adds the key, since we're getting a console warning about the issue.

As Shah said, I'd highly recommend having that in the PR! It's not about the acceptance criteria, more about React's requirements. In most situations it would just work without adding a key attribute, but that doesn't mean we shouldn't.

@labrocadabro @ToobaJamal Let's aim to have that change done before we merge the PR in today's sync.

File: src/views/List.jsx [hunk 1]
```diff
@@ -7,11 +7,9 @@ export function List({ data }) {
...
 			</p>
 			<ul>
-				{/**
-				 * TODO: write some JavaScript that renders the `data` array
-				 * using the `ListItem` component that's imported at the top
-				 * of this file.
-				 */}
+				{data.map((item) => (
+					<ListItem name={item.name} />
+				))}
```
Comment (paramsiddharth) [inline src/views/List.jsx:12]: Please add a `key` attribute to the `ListItem` component. You can use the index as the key.

**Review by paramsiddharth — CHANGES_REQUESTED** [review]

**Comment by labrocadabro** [comment]
@ToobaJamal This is an extremely simple change and I'm a bit busy this morning so can you take care of this yourself?

File: src/views/List.jsx (hunk 1 above)
Comment (ToobaJamal) [reply src/views/List.jsx]: Done that. Please check the changes now :)

**Review by paramsiddharth — APPROVED** [review]
Excellent work! :)

**Review by shahx95 — APPROVED** [review]
Looks good to me! :)

**Pull request merged**